Models for the physical sample

ttung opened the pull request:
### Description
It's still missing the fields from PHA4GE but I think it's easier to throw them in after people agree to the general concepts.

Depends on #30, #42

#### Issue
[ch64743](https://app.clubhouse.io/genepi/stories/space/64743)

### Test plan
Ran migration.

phoenixAja commented on src/py/aspen/database/models.py:
```diff
@@ -46,3 +55,21 @@ class User(base):
+        Group, backref=backref("physical_samples", uselist=True)
+    )
+    private_identifier = Column(String, nullable=False)
+    original_submission = Column(JSON, nullable=False)
```
oh cool a json field

phoenixAja reviewed (approved):
this looks good, i think it captures all the metadata we would need to collect when storing information on physical samples

Commit:
Models for the physical sample

### Description
It's still missing the fields from PHA4GE but I think it's easier to throw them in after people agree to the general concepts.

#### Issue
[ch64743](https://app.clubhouse.io/genepi/stories/space/64743)

### Test plan
Ran migration.

ttung commented:
I updated this with the latest round of revisions, along with some annotations on the db columns.
